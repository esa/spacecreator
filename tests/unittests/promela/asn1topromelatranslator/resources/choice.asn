CHOICES DEFINITIONS ::= BEGIN

Type1 ::= INTEGER(0..3)
Type2 ::= INTEGER(9..13)

SimpleChoice ::= CHOICE 
{
    alt1 Type1,
    alt2 Type2
}

END