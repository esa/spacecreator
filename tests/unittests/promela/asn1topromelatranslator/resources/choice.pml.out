#define SimpleChoice_alt1_PRESENT 1
#define SimpleChoice_alt2_PRESENT 2
inline Type1_generate_value(value)
{
    if
    ::  atomic {
        select (value : 0 .. 3);
    }
    fi;
}
inline Type2_generate_value(value)
{
    if
    ::  atomic {
        select (value : 9 .. 13);
    }
    fi;
}
inline SimpleChoice_generate_value(value)
{
    if
    ::  true;
        Type1_generate_value(value.alt1);
        value.selection = SimpleChoice_alt1_PRESENT;
    ::  true;
        Type2_generate_value(value.alt2);
        value.selection = SimpleChoice_alt2_PRESENT;
    fi;
}
